### ISSUE TITLE
Issues wit Reapit CLI

### ISSUE THREAD

**Issue by nikolai-momot-thirdfort** [reapit/foundations#8033]
**Summary**
HI Folks, I've recently been trying to prepare my team's Reapit app for release and hit some road blocks with the CLI.

1. How do we generate an API Key for the Reapit CLI. The [documentation](https://foundations-documentation.reapit.cloud/app-development/iaas-coming-soon/cli-beta#configure-the-cli) says that there should be a “Pipeline” section on the Apps page but there is none present, does this feature need to be enabled?
<img width="196" alt="Screenshot 2022-10-31 at 15 08 27" src="https://user-images.githubusercontent.com/109656933/199041116-90438f0d-21ff-4bf8-9366-217d7ab56c68.png">
2. The [npm page for reapit/cli](https://www.npmjs.com/package/@reapit/cli) states that an API key can be configured by running reapit config but running any command including reapit config returns an error that states “No config found. Please use the config command before running this command”. Is this a bug?
<img width="549" alt="Screenshot 2022-10-31 at 15 09 24" src="https://user-images.githubusercontent.com/109656933/199041361-89797a7b-58c9-458e-bfeb-a14a780b232c.png">
3. The sample GH Actions workflow described on the [npm page for reapit/cli](https://www.npmjs.com/package/@reapit/cli), states “there is a commit & push action here. This is important to keep the version up to date with deployments. Otherwise deployments will fail after the first successful one.” Can you please clarify what this means? Does that mean that we need to commit something to our main/master branch every time that we attempt to deploy?

4. The CLI section in the documentation states that it is currently in Beta, is there an alternative process of deploying our app on reapit?

**Comment by willmcvay** [comment]
@nikolai-momot-thirdfort You need to request access to the IaaS beta as per the docs https://foundations-documentation.reapit.cloud/app-development/iaas-coming-soon#public-beta Please provide in your email details about your developer organisation and we will get back to you separately.

**Comment by nikolai-momot-thirdfort** [comment]
Thanks @willmcvay I will send an email for access.

Can you please address the other 3 questions in my query?

**Comment by willmcvay** [comment]
Sure;

2. Yes, it looks like a bug - we will look into this
3. Yes, to trigger the action, you will need to generate a relevant Github event
4. No, this is an early beta project we are working on

**Comment by nikolai-momot-thirdfort** [comment]
@willmcvay 

2. Ok, is there a version of reapit/cli that I could use that is stable?

3. I don't think you read my question correctly, I was not referencing what _triggers_ the workflow.
In the GH workflow documented on the the reapit/cli npm page, the last step of the job **is git commit and push**.
Here is the step I am referring to, that is immediately _after_ `pipeline deploy-zip` step:
```
      - name: Commit & Push changes
        uses: actions-js/push@master
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          branch: 'master'
```
The page states "This is important to keep the version up to date with deployments. Otherwise deployments will fail after the first successful one." - this is very vague and doesn't provide much insight.
Can you please clarify _why_ a deploy action would require a commit to the master branch?

4. If this project is in beta, then what is the alternative process of deploying our app on reapit?

**Comment by willmcvay** [comment]
@nikolai-momot-thirdfort 

2. You can try a previous release however, the reapit config command is used to prompt you for an API key. You won't have this at present unless you are part of the beta so a previous version won't help currently. There is a bug and the ticket is in our current sprint.
3. Yes, understood, we will review the docs and publish as part of the bugfix.
4. No, all of our developers self host their applications. While the project is in beta, we are using it internally for dev environments to test our apps pre-release. We would always advise you to do the same for production environments.

**Comment by bashleigh** [comment]
@nikolai-momot-thirdfort 
> The page states "This is important to keep the version up to date with deployments. Otherwise deployments will fail after the first successful one." - this is very vague and doesn't provide much insight.
Can you please clarify why a deploy action would require a commit to the master branch?

This is because the IaaS uses the package.json file's version to create a deployment version. If the semver version is kept at 1.0.0 rollbacks will not work. Therefore to prevent duplication and enable the rollback feature, when running reapit cli in github actions, the action will need to modify the package version to match the IaaS version. The reapit cli bumps the version which is why the github action is required to commit the changes to `package.json` file. Otherwise, deployments will be rejected by IaaS for duplicate version.

This is only required for zip uploads which github actions handles. This isn't a requirement for github commit webhooks.

Hopefully that answers your question?

I've opened a PR #8062  to fix an issue with the reapit/cli. The config command was blocked by an npm version checker function that was added in v2.1.0 I believe? Because there wasn't config and the update checker command uses the same abstract command class, the config condition wasn't met. I've resolved this by making the condition checking for config optional for the update checker command.

**Comment by bashleigh** [comment]
Currently deploying https://github.com/reapit/foundations/releases/tag/%40reapit%2Fcli_v2.1.3 Will be fixed in v2.1.3
